dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/NativeHttpRestClient.java

```diff
@@ -0,0 +1,386 @@
+                    .setRetryHandler(new MyRequestRetryHandler())
+                    .setSSLSocketFactory(socketFactory)
+                    .setConnectionManager(cm)
+                    .build();
+        } else {
+            client = HttpClients.custom().setDefaultCredentialsProvider(credentialsProvider)
+                    .setRetryHandler(new MyRequestRetryHandler())
+                    .setConnectionManager(cm)
+                    .build();
+        }
+        NativeHttpRestClient.url = protocol + hostName + ":" + port;
+
+        List<NamedXContentRegistry.Entry> namedXContentEntries = Collections.emptyList();
+
+        registry = new NamedXContentRegistry(
+                Stream.of(getDefaultNamedXContents().stream(), getProvidedNamedXContents().stream(),
+                        namedXContentEntries.stream())
+                        .flatMap(Function.identity()).collect(toList()));
+    }
+
+    /**
+     * Loads and returns the {@link NamedXContentRegistry.Entry} parsers provided by plugins.
+     */
+    static List<NamedXContentRegistry.Entry> getProvidedNamedXContents() {
+        List<NamedXContentRegistry.Entry> entries = new ArrayList<>();
+        for (NamedXContentProvider service : ServiceLoader.load(NamedXContentProvider.class)) {
+            entries.addAll(service.getNamedXContentParsers());
+        }
+        return entries;
+    }
+
+    public SearchResponse search(SearchRequest searchRequest, RequestOptions options) {
+
+        final HttpGetWithEntity httpGetWithEntity = new HttpGetWithEntity();
+
+        try {
+
+            httpGetWithEntity.setURI(new URI(url  + "/" + searchRequest.indices()[0] + "/_search"));
+
+            httpGetWithEntity.setEntity(createEntity(searchRequest.source(), REQUEST_BODY_CONTENT_TYPE));
+
+            final CloseableHttpResponse response = client.execute(httpGetWithEntity);
+
+            if (response.getStatusLine().getStatusCode() != HttpStatus.SC_OK) {
+                System.err.println("Method failed: " + response.getStatusLine());
+            }else{
+                return parseEntity(response.getEntity(), SearchResponse::fromXContent);
+            }
+
+        } catch (IOException e) {
+            System.err.println("Fatal transport error: " + e.getMessage());
+            e.printStackTrace();
+        } catch (URISyntaxException e) {
+            e.printStackTrace();
+        } finally {
+            // Release the connection.
+            httpGetWithEntity.releaseConnection();
+        }
+
+        return null;
+    }
+
+    HttpEntity createEntity(ToXContent toXContent, XContentType xContentType) throws IOException {
+        return createEntity(toXContent, xContentType, ToXContent.EMPTY_PARAMS);
+    }
+
+    HttpEntity createEntity(ToXContent toXContent, XContentType xContentType, ToXContent.Params toXContentParams)
+            throws IOException {
+        BytesRef source = XContentHelper
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'source' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5121287)